infoblox_client/connector.py

```diff
@@ -439,6 +457,15 @@ def delete_object(self, ref, delete_arguments=None):
         self._validate_authorized(r)
 
         if r.status_code != requests.codes.ok:
+
+            if r.status_code == requests.codes.SERVICE_UNAVAILABLE:
+                raise ib_ex.InfobloxGridTemporaryUnavailable(
+                    response=r.content,
```

Comment:
All the three check are duplicates and can be moved to a separate function itself. It will be easier to unit test also.

Reply:
Done